src/gh/diffCheck/diffCheck/df_vizualization.py

```diff
@@ -0,0 +1,293 @@
+    """
+
+    def __init__(self, name):
+        self.names = name
+        if name == "Jet":
+            self.colors = [
+                Color.FromArgb(0, 0, 255),  # Blue
+                Color.FromArgb(0, 255, 255),  # Cyan
+                Color.FromArgb(0, 255, 0),  # Green
+                Color.FromArgb(255, 255, 0),  # Yellow
+                Color.FromArgb(255, 0, 0),  # Red
+            ]
+        elif name == "Rainbow":
+            self.colors = [
+                Color.FromArgb(127, 0, 255),
+                Color.FromArgb(0, 180, 235),
+                Color.FromArgb(128, 254, 179),
+                Color.FromArgb(255, 178, 96),
+                Color.FromArgb(255, 6, 3)
+            ]
+        elif name == "RdPu":
+            self.colors = [
+                Color.FromArgb(254, 246, 242),
+                Color.FromArgb(251, 196, 191),
+                Color.FromArgb(246, 103, 160),
+                Color.FromArgb(172, 1, 125),
+                Color.FromArgb(76, 0, 106)
+            ]
+        elif name == "Viridis":
+            self.colors = [
+                Color.FromArgb(68, 3, 87),
+                Color.FromArgb(58, 82, 139),
+                Color.FromArgb(32, 144, 140),
+                Color.FromArgb(94, 201, 97),
+                Color.FromArgb(248, 230, 33)
+            ]
+
+
+def interpolate_color(color1, color2, t):
+    """
+    Interpolate between two colors.
+    """
+
+    r = int(color1.R + (color2.R - color1.R) * t)
+    g = int(color1.G + (color2.G - color1.G) * t)
+    b = int(color1.B + (color2.B - color1.B) * t)
+
+    return Color.FromArgb(r, g, b)
+
+
+def value_to_color(value, min_value, max_value, palette):
+    """
+    Map a value to a color based on a colormap.
+    """
+
+    if value < min_value:
+        value = min_value
+    elif value > max_value:
+        value = max_value
+
+    colormap = palette.colors
+
+    # Normalize the value within the range
+    if min_value == max_value:
+        t = 0.5
+    else:
+        t = (value - min_value) / (max_value - min_value)
+
+    # Determine the segment in the colormap
+    n = len(colormap) - 1
+    idx = int(t * n)
+    if idx >= n:
+        idx = n - 1
+    t = (t * n) - idx
+
+    # Interpolate between the two colors
+    color1 = colormap[idx]
+    color2 = colormap[idx + 1]
+
+    return interpolate_color(color1, color2, t)
+
+
+def color_pcd(pcd, values, min_value, max_value, palette):
+    """
+    Colors a point cloud data based on given values and palette.
+    """
+
+    for i, p in enumerate(pcd):
+        # check if values is a list
+        if isinstance(values, list):
+            mapped_color = value_to_color(values[i], min_value, max_value, palette)
+        else:
+            mapped_color = value_to_color(values, min_value, max_value, palette)
+
+        p.Color = mapped_color
+
+    return pcd
+
+
+def color_mesh(mesh, values, min_value, max_value, palette):
```

Comment:
here we should add naming for differentiate `rh` and `df` because we have similar methods in our df objects